app/services/duplicate_case_checker.rb

```diff
@@ -0,0 +1,22 @@
+class DuplicateCaseChecker
+  def self.perform(cases)
+    new(cases).perform
+  end
+
+  def initialize(cases)
+    @cases = cases
+  end
+
+  def perform
+    cases.reject { |kase| case_already_exists?(kase) }
```

Comment:
This is just a personal thing, but I dislike the `c -> k` convention... I'd prefer adding an extra label, either something descriptive like `test_case` or something redundant like `case_record`.